parlai/tasks/qangaroo/agents.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.teachers import DialogTeacher, FixedDialogTeacher
+from .build import build
+
+import json
+import os
+
+class IndexTeacher(FixedDialogTeacher):
+    def __init__(self, opt, shared=None):
+        datapath = self._path(opt)
```

Comment:
just for cleanliness let's have `super().__init__` be the first line before the _path call